### PR TITLE
DPU template view works same way as Canvas view

### LINKED ISSUE #433: Correct handling of invalid configuration in DPU template screen
As reported in https://github.com/UnifiedViews/Plugins/issues/207, there is a different behaviour in menus Pipeline editor and DPU Templates. In menu DPU Templates, when I try to save invalid DPU configuration, after pressing Save button Error message is issued and configuration is reinitialized to last valid configuration. For example if I miss some character but make many changes in configuration, all changes are then lost. In Pipeline editor (pipeline canvas) in DPU config dialog, it works correctly. After pressing Save, error message is issued but invalid configuration stays untouched, so I have chance to correct it.

In case of DPU template screen, if incorrect DPU template conf is specified, the following error occurs and defautl conf is loaded (which causes that user loses whole conf):
screen shot 2015-06-26 at 12 08 51

Solution: In case of adjusting DPU configuration in DPU template screen, it should behave also correctly - if invalid DPU template conf is specified, error is shown, new conf is not saved,  and user can adjust the invalid configuration (he must not lose the invalid configuration, default configuration must not replace his invalid configuration). 

The different behavior may be observed e.g. on t-sparqlUpdate with NO query defined - in case of DPU configuration on pipeline canvas - it behaves correctly - user is requested to correct invalid parts of the configuraiton. In caes of DPU templates it just reports problem and loads default conf

### PR DESCRIPTION
**Pull request by mkl-eea-sk**
Resolve #433

### CODE REVIEW

File: frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java [hunk 1]
```diff
@@ -21,7 +22,7 @@
      * @param dpuWrap
      *            DPU to save.
      */
-    public void saveDPUEventHandler(DPUTemplateWrap dpuWrap);
```
Comment (tomas-knap) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java]: Javadoc has to be updated for both changed methods in DPUPresenter

File: frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java [hunk 2]
```diff
@@ -155,7 +156,7 @@
          * Refresh view.
          */
         public void refresh();
-
+        void treeSetValue(Object oldValue);
```
Comment (tomas-knap) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java]: Javadoc has to be added

File: frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java (hunk 1 above)
Comment (mkl-eea-sk) [inline frontend/src/main/java/cz/cuni/mff/xrg/odcs/frontend/gui/views/dpu/DPUPresenter.java]: are you sure? this interface is MVP used in frontend, it has exactly one implementation and never will have more... it is unfeature once again and you want to update javadocs for something that should be removed at all....

**Comment by tomas-knap** [comment]
Thanks, Ok.

**Comment by mkl-eea-sk** [comment]
javadocs resolved

**Comment by tomas-knap** [comment]
Wait with the merge for approval https://github.com/UnifiedViews/Plugin-DevEnv/pull/57

**Comment by skrchnavy** [comment]
Commits are merged, github wrongly report status of PR #446 as closed instead of Merged.

**Pull request closed without merging**